include/kernel.h

```diff
@@ -3022,6 +3022,26 @@ struct k_delayed_work {
 	struct k_work_q *work_q;
 };
 
+/**
+ * @brief Structure that pairs a work item with a timeout.
+ *
+ * Used with k_work_schedule_for_queue() to allow a work item to be
+ * automatically submitted after a delay.
+ *
+ * No fields in this structure are to be accessed directly.
+ */
+struct k_work_delayable {
+	/* Internal work item that is submitted. */
+	struct k_work work;
+
+	/* Timeout used for delayed submission. */
+	struct _timeout timeout;
```

Comment:
what locking protects this timeout?

Reply:
The same `k_spinlock` in `kernel/work_q.c` that protects the equivalent timeout in the original implementation.